packages/hardhat-core/src/types/network.ts

```diff
@@ -0,0 +1,61 @@
+import { EventEmitter } from "events";
+
```

Comment:
Maybe this module should be `provider.ts` instead of `network.ts`.

Reply:
I'm ok with `provider`, but I think `network` is more friendly and maybe more "future-proof", in the sense that more diverse things can be added in the future.

But, again, both work for me.

Reply:
I named it `network` initially since I thought it might be interesting for future features but it might end up being awkward taking the name over now instead if reserving it for later when it does make sense. I renamed it to `provider` for now.